automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/NotebookPage.scala

```diff
@@ -174,49 +175,51 @@ class NotebookPage(override val url: String)(override implicit val authToken: Au
     await condition isKernelShutdown
   }
 
+  /**
+    * Throw TimeoutException if Kernel is not ready after restart when timeout is reached
+    *
+    * @param timeout
+    */
   def restartKernel(timeout: FiniteDuration = 1 minute): Unit = {
     click on kernelMenu
     click on (await enabled restartKernelSelection)
     click on (await enabled restartKernelConfirmationSelection)
-    waitForIdleKernel(timeout)
-  }
-
-  /**
-    * wait for kernel icon to become idle. default timeout == 1.minute
-    */
-  def waitForIdleKernel(timeout: FiniteDuration = 1.minutes): Unit = {
-    await condition (
-      !cellsAreRunning && isKernelIdle() &&
-        find(id("notification_kernel")).exists(_.underlying.getCssValue("display") == "none"), timeout.toSeconds
-    )
-    await condition (find(id("kernel_indicator_icon")).exists(_.isDisplayed), 2) // little time to check isDisplayed to prevent StaleWebElementException
+    await notVisible restartKernelConfirmationSelection
+    await condition (isKernelReady && kernelNotificationText() == "none", timeout.toSeconds)
   }
 
   /**
     * wait for kernel to become ready. default timeout == 1.minute
+    *
+    * Throw TimeoutException if Kernel is not ready when timeout is reached
     */
-  def awaitReadyKernel(timeout: FiniteDuration = 1.minute): Unit = {
-    try {
-      waitForIdleKernel(timeout)
-    } catch {
-      case _: TimeoutException =>
-        if (isKernelDisconnected()) {
-          restartKernel(timeout)
-        }
+  def waitReadyKernel(timeout: FiniteDuration = 1.minutes): Unit = {
+    Try {
+      await condition (
+        !cellsAreRunning &&
+          isKernelReady() &&
+          kernelNotificationText() == "none", timeout.toSeconds
+      )
+    } recover {
+      case e: TimeoutException => restartKernel(timeout)
```

Comment:
Styling nits: 
1. Right-indent `case e...`
2. Remove the blank line between `recover` and `case`